### ISSUE TITLE
UDA Splits Not Used in domainbed/scripts/train.py During Training

### ISSUE THREAD

**Issue by YaraLiao** [Luodian/Generalizable-Mixture-of-Experts#10]
I noticed that in the code domainbed/scripts/train.py, the uda_splits are not utilized during training. Could you clarify if there is an updated version of the code that integrates the uda_splits into the training step? Additionally, does the updated version unify the step and the uda_split settings?

For the leave-one-domain-out setting, are the reported results based solely on the test set (i.e., the domain used as the test set) with cross-validation (leaving each domain out as the test set in turn)? I am a bit confused because I couldn’t find a validation set in the code. Could you clarify this part?

Looking forward to your response. Thank you!

**Comment by Lixiaomemg** [comment]
Hello, I would like to ask, your running command is
python3 -m domainbed.scripts.train
--data_dir=./domainbed/data/OfficeHome/
--algorithm GMOE
--dataset OfficeHome
--test_env 2
Are there any other changes in the code or command?
I don’t know why after I run it, I run
python -m domainbed.scripts.collect_results
--input_dir=/my/sweep/output/path
The result is very different from that presented in the paper.
For example, on the PACS dataset, my running result is only 40.6.

**Comment by YaraLiao** [comment]
I didn’t run the PACS dataset, but my results in VLCS dataset align with the results in the paper.
I wonder do you use the pretrained model copts?

**Comment by Lixiaomemg** [comment]
Can I ask if you run domainbed.scripts.train first and then domainbed.scripts.collect_results? Also, what version of tutel is in your environment?
